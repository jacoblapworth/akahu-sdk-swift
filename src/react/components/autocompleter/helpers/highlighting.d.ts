/**
 * Find all case-insensitive matches of a sub-string in a string and replace them with the
 * output of decorateFn. For example, if you wanted to italicize all instances of 'ca' in
 * 'Candy cane', you could:
 *
 * decorateSubStr('Candy cane', 'ca', (subStr) => (<i>{subStr}</i>));
 *
 * // => [<i>Ca</i>, 'ndy ', <i>ca</i>, 'ne'];
 */
export const decorateSubStr: <T>(
  str: string,
  searchStr: string,
  decorateFn: (str: string, key: string) => T,
) => string | Array<string | T>;

/**
 * Decorator function that can be used with decorateSubStr to bold matching
 * selections by wrapping them in a strong tag.
 */
export const boldMatch: (str: string, id: string) => React.Component;
