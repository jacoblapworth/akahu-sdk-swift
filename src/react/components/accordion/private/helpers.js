import verge from 'verge';
import {calcSpaceRight, calcSpaceLeft} from '../../positioning/private/dom-helpers';

const getSpacesHorizontal = rect => ({
	spaceLeft: calcSpaceLeft(rect),
	spaceRight: calcSpaceRight(rect),
});

const hasEnoughSpaceHorizontal = ({ spaceLeft, spaceRight }) => {
	const MARGIN_POP = 8;
	const TOLERANCE = 1;
	const WIN_SCROLL_BAR = 17;

	return (
		spaceLeft > MARGIN_POP + TOLERANCE && spaceRight > MARGIN_POP + TOLERANCE + WIN_SCROLL_BAR
	);
};

export const shouldAccordionPop = rect => hasEnoughSpaceHorizontal(getSpacesHorizontal(rect));
export const getRectangle = rect => verge.rectangle(rect);
