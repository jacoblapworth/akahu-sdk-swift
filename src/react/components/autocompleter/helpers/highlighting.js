import React from 'react';
import { v4 as uuidv4 } from 'uuid';

/**
 * Find all case-insensitive matches of a sub-string in a string and replace them with the
 * output of decorateFn. For example, if you wanted to italicize all instances of 'ca' in
 * 'Candy cane', you could:
 * decorateSubStr('Candy cane', 'ca', (subStr) => (<i>{subStr}</i>));
 * // => [<i>Ca</i>, 'ndy ', <i>ca</i>, 'ne'];
 *
 * @public
 * @param {string} str - The string being searched
 * @param {string} searchStr - The sub-string which needs to be decorated
 * @param {function} decorateFn - Function which will map the string to whatever you want
 * @returns {Array.<string|JSX>|string}
 */
export const decorateSubStr = (str, searchStr, decorateFn) => {
  // If the search string is the same as the result, return a decorated full match
  if (str === searchStr) {
    // v1() creates a unique time based key
    return decorateFn(str, uuidv4());
  }

  // If there's no search string or its greater than a match there's no need to
  // decorate in this case.
  if (!str || !searchStr || searchStr.length > str.length) {
    return str;
  }
  const matchRegex = new RegExp(searchStr, 'ig');
  const matches = str.match(matchRegex);
  if (!matches) {
    return str;
  }

  // Decorate matches by splitting 'str' using the matchRegex, creating an array which can contain
  // all string parts and matches, then merging the items in the split string with the decorated
  // matches.
  const splitString = str.split(matchRegex);
  const result = [];
  let resultIdx = 0;
  for (let i = 0; i < splitString.length; i += 1) {
    if (splitString[i]) {
      result[resultIdx] = splitString[i];
      resultIdx += 1;
    }
    if (matches[i]) {
      result[resultIdx] = decorateFn(matches[i], uuidv4());
      resultIdx += 1;
    }
  }
  return result;
};

/**
 * Decorator function that can be used with decorateSubStr to bold matching
 * selections by wrapping them in a strong tag.
 *
 * @public
 * @param {string} str
 * @param {string} id
 *
 * @returns {React.Component}
 */
export const boldMatch = (str, id) => <strong key={id}>{str}</strong>;
