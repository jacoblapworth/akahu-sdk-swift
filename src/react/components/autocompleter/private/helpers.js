
/**
 * Simple predicated used to determine if the ListBox's root DOM node can actually be focused.
 *
 * @public
 * @param {HTMLElement} node
 *
 * @returns {boolean}
 */
export const isVisible = node => node && window.getComputedStyle(node).visibility !== 'hidden';

/**
 * Attempts to run the passed predicate, if successful it will run the callback method and clear. If unsuccessful, it
 * will run several times over five seconds. If it's still unsuccessful after that it will try again several times over
 * a half second. If it's still not successful it will stop trying.
 *
 * @public
 * @param {Function} predicate
 * @param {Function} callback
 */
export const intervalRunner = (predicate, callback) => {
	if (predicate()) {
		callback();
	} else {
		let counter = 0;
		let interval = 0;
		const maxCount = 5;
		const delay = 100;
		const checker = function () {
			const check = predicate();
			if (check) {
				callback();
			}
			if (check|| counter > maxCount) {
				clearInterval(interval);
			}
			counter++;
		};

		interval = setInterval(checker, delay);
	}
};

export const dropdownSizeClasses = {
	small: 'ac-dropdown-small',
	medium: 'ac-dropdown-medium',
	large: 'ac-dropdown-large',
	xlarge: 'ac-dropdown-xlarge'
};
