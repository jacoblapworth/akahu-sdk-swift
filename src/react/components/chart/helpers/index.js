import getGroupPosition from './groupposition';
import createChartPadding from './chartpadding';

const alwaysPositive = value => Math.max(0, value) || 0;
const testIsCloseEnough = (next, previous, threshold = 2) => (
	next > previous - threshold && next < previous + threshold
);

export {getGroupPosition as default, createChartPadding, alwaysPositive, testIsCloseEnough};
