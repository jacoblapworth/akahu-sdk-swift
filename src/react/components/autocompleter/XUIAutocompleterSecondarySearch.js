import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import XUIAutocompleterTextInput from './XUIAutocompleterTextInput';
import DropDown from '../dropdown/DropDown';
import DropDownToggled from '../dropdown/DropDownToggled';
import searchPath from '@xero/xui-icon/icons/search'
import XUITextInputSideElement from '../textInput/XUITextInputSideElement';
import XUIIcon from '../icon/XUIIcon';
import {ns} from "../helpers/xuiClassNamespace";

import { intervalRunner, isVisible } from './private/helpers';

export default class XUIAutocompleterSecondarySearch extends PureComponent {
	constructor(props) {
		super(props);
		this.openDropDown = this.openDropDown.bind(this);
		this.closeDropDown = this.closeDropDown.bind(this);
		this.onOpen = this.onOpen.bind(this);
		this.highlightItem = this.highlightItem.bind(this);
	}

	/**
	 * Set the state as not hidden in order to toggle the list open.
	 *
	 * @public
	 */
	openDropDown() {
		this.ddt.openDropDown();
	}

	/**
	 * Set the state as hidden in order to toggle the list closed.
	 *
	 * @public
	 */
	closeDropDown() {
		this.ddt.closeDropDown();
	}

	/**
	 * Manually highlight an item in the list for selection.
	 *
	 * @public
	 */
	highlightItem(item) {
		this.dropdown.highlightItem(item);
	}

	/**
	 * Focuses the autocompleter input before calling props.onOpen
	 *
	 * @public
	 */
	onOpen() {
		this.focusInput();
		this.props.onOpen && this.props.onOpen();
	}

	/**
	 * Focus the XUIAutocompleterInput input element, if visible.
	 *
	 * @public
	 */
	focusInput() {
		const inputDOM = this.input;
		const isInputRendered = () => isVisible(inputDOM);
		const setter = () => {
			inputDOM.focus();
		};
		intervalRunner(isInputRendered, setter);
	}

	render() {
		const completer = this;
		const props = completer.props;

		let listQaHook = null;
		let containerQaHook = null;
		if (props.qaHook) {
			listQaHook = `${props.qaHook}--list`;
			containerQaHook = `${props.qaHook}--container`;
		}
		const dropdownClasses = cn(
			props.dropdownSize ? '' : `${ns}-u-fullwidth`,
			props.dropdownClassName,
		);
		const searchItem = (
			<div className={`${ns}-dropdown--header-container`}>
					<XUIAutocompleterTextInput
						value={props.searchValue}
						leftElement={
							<XUITextInputSideElement>
								<XUIIcon path={searchPath} />
							</XUITextInputSideElement>
						}
						placeholder={props.placeholder}
						throttleInterval={props.searchThrottleInterval}
						onChange={props.onSearch}
						inputRef={c => completer.input = c}
						textInputComponentProps={{
							isBorderlessSolid: true
						}}
						inputProps={{
							id: props.inputId
						}}
					/>
			</div>
		);
		const dropdownToggledClasses = props.dropdownSize ? '' : `${ns}-u-fullwidth`;
		const dropdown = (
			<DropDown
				ref={d => completer.dropdown = d}
				ignoreKeyboardEvents={[32,37,39]} /* Space doesn't select in an autocompleter; left and right arrow keys should move cursor in the input */
				hasKeyboardEvents={false}
				onSelect={props.onOptionSelect}
				id={props.id}
				className={dropdownClasses}
				qaHook={listQaHook}
				size={props.dropdownSize}
				fixedWidth={props.dropdownFixedWidth}
				header={searchItem}
				footer={props.footer}
				restrictFocus={props.restrictFocus}
			>
			{props.children}
		</DropDown>);

		return (
			<div
				ref={c => completer.rootNode = c}
				className={props.className}
				data-automationid={containerQaHook}
			>
				<DropDownToggled
					ref={c => completer.ddt = c}
					trigger={props.trigger}
					dropdown={dropdown}
					onOpen={completer.onOpen}
					onClose={props.onClose}
					closeOnSelect={props.closeOnSelect}
					className={dropdownToggledClasses}
					matchTriggerWidth={props.matchTriggerWidth}
				/>
			</div>
		);
	}
}

XUIAutocompleterSecondarySearch.propTypes = {
	/** Callback to handle when an option has been selected from the dropdown */
	onOptionSelect: PropTypes.func,

	/**
	 * When set to true a loader will be displayed instead of the picklist items.
	 * State for this should be managed externally and it's defaulted to false.
	 */
	loading: PropTypes.bool,

	/** ID to be added to the dropdown element of the completer */
	id: PropTypes.string,

	/** Value that should be inside the input. */
	searchValue: PropTypes.string,

	/** CSS class(es) to go on the wrapping DOM node */
	className: PropTypes.string,

	/** CSS class(es) to go on the dropdown list */
	dropdownClassName: PropTypes.string,

	/** CSS class(es) to go on the input */
	inputClassName: PropTypes.string,

	/** Placeholder for the input */
	placeholder: PropTypes.string,

	/** A set of pills to show above the input.  Useful for showing what was selected in a multi-select */
	pills: PropTypes.oneOfType([PropTypes.element, PropTypes.arrayOf(PropTypes.element)]),

	/** Callback for when the list opens */
	onOpen: PropTypes.func,

	/** Callback for when the list closes */
	onClose: PropTypes.func,

	/** Callback for when the user types into the search box */
	onSearch: PropTypes.func,

	/** If you want to throttle the input's onChange handler, put the throttle interval here */
	searchThrottleInterval: PropTypes.number,

	/** Maps to the 'size' property of the dropdown component. */
	dropdownSize: PropTypes.oneOf(['small', 'medium', 'large', 'xlarge']),

	/** Maps to the `closeOnSelect` property of the DropDownToggled component. */
	closeOnSelect: PropTypes.bool,

	/** When set to true the dropdown will automatically open when the input is given focus. */
	openOnFocus: PropTypes.bool,

	/** Will be passed directly down to the DropDownToggled component as the main trigger. */
	trigger: PropTypes.element.isRequired,

	/** ID to be applied to the XUIAutocompleterInput component. */
	inputId: PropTypes.string,

	/** Force the desktop user experience, even if the viewport is narrow enough for mobile. */
	forceDesktop: PropTypes.bool,

	/** If a size is set, this will force the dropdown to that size instead of setting it as a max width. */
	dropdownFixedWidth: PropTypes.bool,

	/** Whether focus should be restricted to the dropdown while it's open. */
	restrictFocus: PropTypes.bool,

	/**
	 * Setting to true will allow the dropdown's width to be set dependent of the trigger width.
	 * Note: Setting this to true will override any size prop on DropDown. XUI design has also decided
	 * to keep a minimum width on the dropdown, so dropdown may not match the width of narrow triggers.
	 */
	matchTriggerWidth: PropTypes.bool,

	/** A Footer element can be added. */
	footer: PropTypes.element,

	qaHook: PropTypes.string,
	children: PropTypes.node,
};

XUIAutocompleterSecondarySearch.defaultProps = {
	loading: false,
	searchThrottleInterval: 0,
	openOnFocus: false,
	inputId: 'secondary_search_input',
	restrictFocus: true,
};
