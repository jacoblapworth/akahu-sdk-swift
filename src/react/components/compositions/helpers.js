import cn from 'classnames';
import { ns } from '../helpers/xuiClassNamespace';

const baseCompositionClass = `${ns}-composition`;

export const retainValues = {
	medium: 'medium',
	small: 'small',
};

export function buildGlobalCompositionClasses({
	isInfinite,
	hasGridGap,
	hasAutoSpaceAround,
	hasAutoColumnWidths,
}) {
	return cn(
		baseCompositionClass,
		!isInfinite && `${baseCompositionClass}-is-finite`,
		hasGridGap && `${baseCompositionClass}-has-grid-gap`,
		hasAutoSpaceAround && `${baseCompositionClass}-has-auto-space-around`,
		hasAutoColumnWidths && `${baseCompositionClass}-has-auto-widths`,
	);
}

export default baseCompositionClass;
