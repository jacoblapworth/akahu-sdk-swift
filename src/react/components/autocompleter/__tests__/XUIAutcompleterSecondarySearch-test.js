import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';
import { nanoid } from 'nanoid';
import { axe, toHaveNoViolations } from 'jest-axe';
import XUIAutocompleterSecondarySearch from '../XUIAutocompleterSecondarySearch';
import XUIDropdownFooter from '../../dropdown/XUIDropdownFooter';
import XUIButton from '../../button/XUIButton';
import XUILoader from '../../loader/XUILoader';
import XUIPicklist from '../../picklist/XUIPicklist';
import XUIPickitem from '../../picklist/XUIPickitem';
import Pill from '../../pill/XUIPill';
import { eventKeyValues } from '../../helpers/reactKeyHandler';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import wait from '../../../helpers/wait';

jest.mock('nanoid');
nanoid.mockImplementation(() => 'xui-testDropdownId');

Enzyme.configure({ adapter: new Adapter() });
expect.extend(toHaveNoViolations);

const trigger = <button type="button">trigger</button>;
const createComponent = props => (
  <XUIAutocompleterSecondarySearch
    onSearch={() => {}}
    restrictFocus={false}
    trigger={trigger}
    {...props}
  >
    <XUIPicklist>
      <XUIPickitem id="1">item 1</XUIPickitem>
      <XUIPickitem id="2">item 2</XUIPickitem>
    </XUIPicklist>
  </XUIAutocompleterSecondarySearch>
);

describe('<XUIAutoCompleterSecondarySearch />', () => {
  let wrapper;

  it('should render an automation id when a qahook is passed', () => {
    const automationid = renderer.create(createComponent({ qaHook: 'secondary-search' }));

    expect(automationid).toMatchSnapshot();
  });

  it('should render extra classes when passed to the className prop', () => {
    const classComp = renderer.create(createComponent({ className: 'secondary-search-class' }));

    expect(classComp).toMatchSnapshot();
  });

  it('should render extra classes to the input when passed to the inputclassName prop', () => {
    const classComp = renderer.create(createComponent({ inputclassName: 'input-class' }));

    expect(classComp).toMatchSnapshot();
  });

  it('should render extra classes to the dropdown when passed to the dropdownClassName prop', () => {
    const classComp = renderer.create(createComponent({ dropdownClassName: 'dropdown-class' }));

    expect(classComp).toMatchSnapshot();
  });

  it('displays a XUILoader when isLoading is true', () => {
    wrapper = mount(createComponent({ isLoading: true }));

    expect(wrapper.find(XUILoader)).toBeDefined();
    expect(wrapper.prop('isLoading')).toBeTruthy();
  });

  it('renders with isLoading as false by default', () => {
    wrapper = mount(createComponent());
    expect(wrapper.prop('isLoading')).toBeFalsy();
  });

  it('should render a placeholder inside the input when passed in', () => {
    const comp = renderer.create(createComponent({ placeholder: 'Search' }));

    expect(comp).toMatchSnapshot();
  });

  it('should render as disabled when isDisabled prop is true', () => {
    const disabled = renderer.create(createComponent({ isDisabled: true }));

    expect(disabled).toMatchSnapshot();
  });

  it('should render an id on the dropdown node when passed in the id prop', () => {
    const classComp = renderer.create(createComponent({ id: 'test-id' }));

    expect(classComp).toMatchSnapshot();
  });

  it('should render an id on the input when a value is passed to the inputId prop', () => {
    const inputId = renderer.create(createComponent({ inputId: 'test-inputId' }));

    expect(inputId).toMatchSnapshot();
  });

  it('renders pills as children passed in through the pills prop', () => {
    wrapper = mount(createComponent({ pills: <Pill value="ABC" /> }));

    expect(wrapper.find(Pill)).toBeDefined();
  });

  it('should call the onOpen callback when the dropdown is opened', () => {
    const onOpen = jest.fn();
    wrapper = mount(createComponent({ onOpen, searchValue: 'on open', onSearch: onOpen }));

    wrapper.instance().openDropdown();

    expect(onOpen.mock.calls.length).toEqual(1);
  });

  // Skipped until we can work around rending content with portals in tests.
  it.skip('should call onSearch when a search is done', () => {
    const onSearch = jest.fn();
    const searchComp = mount(createComponent({ onSearch, value: 'old value' }));

    // Simulate opening the dropdown
    searchComp.find('button').simulate('click');

    // Simulate changigns the value of the input field
    searchComp.find('input').simulate('change', {
      target: {
        value: 'new value',
      },
    });

    expect(onSearch.mock.calls.length).toEqual(1);
  });

  it('should pass the isBlock prop to the XUIDropdownToggled element', () => {
    const searchComp = mount(createComponent({ isBlock: true }));

    const dropdownToggledComp = searchComp.find('XUIDropdownToggled');

    expect(dropdownToggledComp.prop('isBlock')).toBeTruthy();
  });

  it('should open the dropdown when space is pressed', () => {
    wrapper = mount(createComponent());
    const triggerButton = wrapper.find('button');
    triggerButton.simulate('keyDown', { key: eventKeyValues.space, keyCode: 32, which: 32 });

    expect(wrapper.instance().ddt.current.state.isHidden).toBeFalsy();
  });

  it('should pass accessibility testing', async () => {
    const component = mount(createComponent());
    const results = await axe(component.html());
    expect(results).toHaveNoViolations();
  });

  describe('New focus behaviour', () => {
    describe('without a footer', () => {
      test('pressing enter from the trigger opens the dropdown and focuses the input inside the dropdown', async () => {
        // Arrange
        render(
          createComponent({
            qaHook: 'autocompleterSS',
            trigger: <XUIButton qaHook="trigger">Trigger</XUIButton>,
            useNewFocusBehaviour: true,
          }),
        );

        // Act
        userEvent.click(screen.getByTestId('trigger'));
        await wait(100);

        // Assert
        expect(screen.queryByTestId('autocompleterSS--list')).toBeInTheDocument();
        expect(screen.getByTestId('autocompleterSS--input--input')).toHaveFocus();
      });

      test('tabbing from an open autocompleter secondary search closes the dropdown and focuses the next element', async () => {
        // Arrange
        render(
          <>
            {createComponent({
              qaHook: 'autocompleterSS',
              trigger: <XUIButton qaHook="trigger">Trigger</XUIButton>,
              useNewFocusBehaviour: true,
            })}
            <XUIButton qaHook="nextElement">Next Element</XUIButton>
          </>,
        );

        // Act
        userEvent.click(screen.getByTestId('trigger'));
        await wait(100);
        userEvent.tab();

        // Assert
        expect(screen.getByTestId('nextElement')).toHaveFocus();
        // @TODO XUI-2882 Un-comment this line
        // expect(screen.queryByTestId('autocompleterSS--list')).not.toBeInTheDocument();
      });

      test('shift tabbing from an open autocompleter secondary search closes the dropdown and focuses the trigger', async () => {
        // Arrange
        render(
          createComponent({
            trigger: <XUIButton qaHook="trigger">Trigger</XUIButton>,
            qaHook: 'autocompleterSS',
            useNewFocusBehaviour: true,
          }),
        );

        // Act
        userEvent.click(screen.getByTestId('trigger'));
        await wait(100);
        userEvent.tab({ shift: true });

        // // Assert
        expect(screen.getByTestId('trigger')).toHaveFocus();
        // @TODO XUI-2882 Un-comment this line
        // expect(screen.queryByTestId('autocompleterSS--list')).not.toBeInTheDocument();
      });
    });

    describe('with a footer', () => {
      const footerComponent = (
        <XUIDropdownFooter
          pickItems={
            <XUIPickitem id="footer" qaHook="autocompleterSSFooter">
              Footer
            </XUIPickitem>
          }
        />
      );

      test('pressing enter from the trigger opens the dropdown and focuses the input inside the dropdown', async () => {
        // Arrange
        render(
          createComponent({
            footer: footerComponent,
            qaHook: 'autocompleterSS',
            trigger: <XUIButton qaHook="trigger">Trigger</XUIButton>,
            useNewFocusBehaviour: true,
          }),
        );

        // Act
        userEvent.click(screen.getByTestId('trigger'));
        await wait(100);

        // Assert
        expect(screen.queryByTestId('autocompleterSS--list')).toBeInTheDocument();
        expect(screen.getByTestId('autocompleterSS--input--input')).toHaveFocus();
      });

      test('tabbing from an open autocompleter secondary search focuses the dropdown footer', async () => {
        // Arrange
        render(
          createComponent({
            footer: footerComponent,
            qaHook: 'autocompleterSS',
            trigger: <XUIButton qaHook="trigger">Trigger</XUIButton>,
            useNewFocusBehaviour: true,
          }),
        );

        // Act
        userEvent.click(screen.getByTestId('trigger'));
        await wait(100);
        userEvent.tab();

        // Assert
        expect(screen.getByTestId('autocompleterSSFooter--body')).toHaveFocus();
      });

      test('tabbing from an autocompleter secondary search footer closes the dropdown and focuses the next element', async () => {
        // Arrange
        render(
          <>
            {createComponent({
              footer: footerComponent,
              qaHook: 'autocompleterSS',
              trigger: <XUIButton qaHook="trigger">Trigger</XUIButton>,
              useNewFocusBehaviour: true,
            })}
            <XUIButton qaHook="nextElement">Next Element</XUIButton>
          </>,
        );

        // Act
        userEvent.click(screen.getByTestId('trigger'));
        await wait(100);
        userEvent.tab();
        userEvent.tab();

        // Assert
        expect(screen.getByTestId('nextElement')).toHaveFocus();
        // @TODO XUI-2882 Un-comment this line
        // expect(screen.queryByTestId('autocompleterSS--list')).not.toBeInTheDocument();
      });

      test('shift tabbing from an autocompleter secondary search footer focuses the input inside the dropdown', async () => {
        // Arrange
        render(
          createComponent({
            footer: footerComponent,
            qaHook: 'autocompleterSS',
            trigger: <XUIButton qaHook="trigger">Trigger</XUIButton>,
            useNewFocusBehaviour: true,
          }),
        );

        // Act
        userEvent.click(screen.getByTestId('trigger'));
        await wait(100);
        userEvent.tab();
        userEvent.tab({ shift: true });

        // Assert
        expect(screen.getByTestId('autocompleterSS--input--input')).toHaveFocus();
      });

      test('shift tabbing from an open autocompleter secondary search closes the dropdown and focuses the input trigger', async () => {
        // Arrange
        render(
          createComponent({
            footer: footerComponent,
            qaHook: 'autocompleterSS',
            trigger: <XUIButton qaHook="trigger">Trigger</XUIButton>,
            useNewFocusBehaviour: true,
          }),
        );

        // Act
        userEvent.click(screen.getByTestId('trigger'));
        await wait(100);
        userEvent.tab({ shift: true });

        // Assert
        expect(screen.getByTestId('trigger')).toHaveFocus();
        // @TODO XUI-2882 Un-comment this line
        // expect(screen.queryByTestId('autocompleterSS--list')).not.toBeInTheDocument();
      });
    });
  });
});
