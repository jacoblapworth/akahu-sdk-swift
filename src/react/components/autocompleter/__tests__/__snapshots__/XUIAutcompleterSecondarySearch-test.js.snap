// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XUIAutoCompleterSecondarySearch /> should render a placeholder inside the input when passed in 1`] = `
<div
  className={undefined}
  data-automationid={null}
>
  <div
    aria-expanded={undefined}
    aria-owns={undefined}
    className="xui-dropdown-fullwidth"
    data-automationid={undefined}
    data-ref="toggled-wrapper"
    role="presentation"
  >
    <button
      aria-activedescendant={null}
      aria-controls={undefined}
      aria-haspopup="listbox"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      trigger
    </button>
  </div>
</div>
`;

exports[`<XUIAutoCompleterSecondarySearch /> should render an automation id when a qahook is passed 1`] = `
<div
  className={undefined}
  data-automationid="secondary-search--container"
>
  <div
    aria-expanded={undefined}
    aria-owns={undefined}
    className="xui-dropdown-fullwidth"
    data-automationid={undefined}
    data-ref="toggled-wrapper"
    role="presentation"
  >
    <button
      aria-activedescendant={null}
      aria-controls={undefined}
      aria-haspopup="listbox"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      trigger
    </button>
  </div>
</div>
`;

exports[`<XUIAutoCompleterSecondarySearch /> should render an id on the dropdown node when passed in the id prop 1`] = `
<div
  className={undefined}
  data-automationid={null}
>
  <div
    aria-expanded={undefined}
    aria-owns={undefined}
    className="xui-dropdown-fullwidth"
    data-automationid={undefined}
    data-ref="toggled-wrapper"
    role="presentation"
  >
    <button
      aria-activedescendant={null}
      aria-controls={undefined}
      aria-haspopup="listbox"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      trigger
    </button>
  </div>
</div>
`;

exports[`<XUIAutoCompleterSecondarySearch /> should render an id on the input when a value is passed to the inputId prop 1`] = `
<div
  className={undefined}
  data-automationid={null}
>
  <div
    aria-expanded={undefined}
    aria-owns={undefined}
    className="xui-dropdown-fullwidth"
    data-automationid={undefined}
    data-ref="toggled-wrapper"
    role="presentation"
  >
    <button
      aria-activedescendant={null}
      aria-controls={undefined}
      aria-haspopup="listbox"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      trigger
    </button>
  </div>
</div>
`;

exports[`<XUIAutoCompleterSecondarySearch /> should render as disabled when isDisabled prop is true 1`] = `
<div
  className={undefined}
  data-automationid={null}
>
  <div
    aria-expanded={undefined}
    aria-owns={undefined}
    className="xui-dropdown-fullwidth"
    data-automationid={undefined}
    data-ref="toggled-wrapper"
    role="presentation"
  >
    <button
      aria-activedescendant={null}
      aria-controls={undefined}
      aria-haspopup="listbox"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      trigger
    </button>
  </div>
</div>
`;

exports[`<XUIAutoCompleterSecondarySearch /> should render extra classes to the dropdown when passed to the dropdownClassName prop 1`] = `
<div
  className={undefined}
  data-automationid={null}
>
  <div
    aria-expanded={undefined}
    aria-owns={undefined}
    className="xui-dropdown-fullwidth"
    data-automationid={undefined}
    data-ref="toggled-wrapper"
    role="presentation"
  >
    <button
      aria-activedescendant={null}
      aria-controls={undefined}
      aria-haspopup="listbox"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      trigger
    </button>
  </div>
</div>
`;

exports[`<XUIAutoCompleterSecondarySearch /> should render extra classes to the input when passed to the inputclassName prop 1`] = `
<div
  className={undefined}
  data-automationid={null}
>
  <div
    aria-expanded={undefined}
    aria-owns={undefined}
    className="xui-dropdown-fullwidth"
    data-automationid={undefined}
    data-ref="toggled-wrapper"
    role="presentation"
  >
    <button
      aria-activedescendant={null}
      aria-controls={undefined}
      aria-haspopup="listbox"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      trigger
    </button>
  </div>
</div>
`;

exports[`<XUIAutoCompleterSecondarySearch /> should render extra classes when passed to the className prop 1`] = `
<div
  className="secondary-search-class"
  data-automationid={null}
>
  <div
    aria-expanded={undefined}
    aria-owns={undefined}
    className="xui-dropdown-fullwidth"
    data-automationid={undefined}
    data-ref="toggled-wrapper"
    role="presentation"
  >
    <button
      aria-activedescendant={null}
      aria-controls={undefined}
      aria-haspopup="listbox"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      trigger
    </button>
  </div>
</div>
`;
